### PR TITLE
Allow association of work item and iteration (#714 and #716)

### PR DESCRIPTION
**Pull request by nimishamukherjee**
Allow association of work item and iteration (#714 and #716)

### CODE REVIEW

File: src/app/iteration/iteration.service.ts [hunk 1]
```diff
@@ -143,12 +144,13 @@ export class IterationService {
     .then((response) => {
       let spaces = response.json().data;
       if (spaces.length) {
+        this.iterationUrl = spaces[0].relationships.iterations.links.related
```
Comment (vikram-raj) [inline src/app/iteration/iteration.service.ts]: Semicolon missing.

**Comment by vikram-raj** [comment]
@nimishamukherjee We need to handle long iteration name in the dropdown and also in detail page.

File: src/app/work-item/common.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    return this.iterationService.getSpaces()
+      .then((data) => {
+        this.iterationService.getIterations(data.relationships.iterations.links.related)
+        .then(iterations =>  iterations)
+        .catch ((e) => {
+          console.log('Some error has occured', e);
+        })
+      })
+      .catch ((err) => {
+        console.log('Spcae not found');
```
Comment (vikram-raj) [inline src/app/work-item/common.resolver.ts]: Typo in space.

**Comment by nimishamukherjee** [comment]
@vikram-raj created issue #909 for handling long iteration names.

File: src/app/iteration/iteration.service.ts [hunk 3]
```diff
@@ -8,6 +8,7 @@ import { Injectable } from '@angular/core';
 export class IterationService {
   public iterations: IterationModel[] = [];
   private headers = new Headers({'Content-Type': 'application/json'});
+  private iterationUrl: string = '';
```
Comment (sanbornsen) [inline src/app/iteration/iteration.service.ts]: Are we using it anywhere?

File: src/app/iteration/iteration.service.ts [hunk 4]
```diff
@@ -143,12 +144,13 @@ export class IterationService {
     .then((response) => {
       let spaces = response.json().data;
       if (spaces.length) {
+        this.iterationUrl = spaces[0].relationships.iterations.links.related;
```
Comment (sanbornsen) [inline src/app/iteration/iteration.service.ts]: I see an assignment but no use of this variable!!

**Pull request merged**